public/index.php

```diff
@@ -8,10 +8,10 @@
 }
 
 header('Content-Type: text/xml; charset=utf-8');
-$base_url = 'http://' . $_SERVER['HTTP_HOST'] . dirname($_SERVER['SCRIPT_NAME']);
+$base_url = http_https() . $_SERVER['SERVER_NAME'] . dirname($_SERVER['SCRIPT_NAME']);
 echo "<" . "?xml version='1.0' encoding='utf-8' standalone='yes'?>";
 ?>
-<service xml:base="<?= $base_url ?>/"
```

Comment:
Rather than removing the slash here, perhaps we should do `rtrim($base_url, '/')` (to remove any existing trailing slash) and then always append `/`.